frontend/src/mapSquareView/mapSquareView.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import Navbar from '../about/Navbar';
+import { Footer } from '../UILibrary/components';
+
+export class MapSquareView extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            loading: true,
+            map_square_data: null,
+        };
+    }
+
+    async componentDidMount() {
+        try {
+            const mapSquareId = window.location.pathname.split('/')[2];
+            const response = await fetch(`/api/map_square/${mapSquareId}/`);
+            if (!response.ok) {
+                this.setState({ loading: false });
+            } else {
+                const map_square_data = await response.json();
+                this.setState({
+                    map_square_data,
+                    loading: false,
+                });
+            }
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+
+    render() {
+        if (this.state.loading) {
```

Comment:
This "loading" infrastructure is starting to look like something we can factor out into its own component.